src/main/java/org/broadinstitute/consent/http/db/SamDAO.java

```diff
@@ -54,10 +54,17 @@ public SamDAO(HttpClientUtil clientUtil, ServicesConfiguration configuration) {
   public List<ResourceType> getResourceTypes(AuthUser authUser) throws Exception {
     GenericUrl genericUrl = new GenericUrl(configuration.getV1ResourceTypesUrl());
     HttpRequest request = clientUtil.buildGetRequest(genericUrl, authUser);
-    HttpResponse response = executeRequest(request);
-    if (!response.isSuccessStatusCode()) {
-      logException("Error getting resource types from Sam: " + response.getStatusMessage(),
-          new ServerErrorException(response.getStatusMessage(), response.getStatusCode()));
+    HttpResponse response;
+    try {
+      response = executeRequest(request);
+      if (!response.isSuccessStatusCode()) {
+        logException("Error getting resource types from Sam: " + response.getStatusMessage(),
+            new ServerErrorException(response.getStatusMessage(), response.getStatusCode()));
+      }
+    } catch (ServerErrorException e) {
```

Comment:
It looks like all `SamDAO` APIs are called from a single class, `SamService`, and its members have the same signature as the defaults class. Why not wrap the DAO calls in `SamService` instead?

Reply:
This separation style is something we've been doing in Terra services forever. I'm not inclined to tackle this as a refactor at this point. I'm not opposed to having a larger conversation around this.

Reply:
I don't think what I'm suggesting is a refactor of the code. To be clear (and to mix together all my suggestions) this is what I'm thinking.
```
public class SamDefaults {

  public List<ResourceType> getDefaultResourceTypes(AuthUser authUser) {
    return List.of();
  }

  public UserStatusInfo getDefaultUserStatusInfo(AuthUser authUser) {
...
```
```
public class SamService implements ConsentLogger {

  private final SamDAO samDAO;

  private final SamDefaults samDefaults = new SamDefaults();

  @Inject
  public SamService(SamDAO samDAO) {
    this.samDAO = samDAO;
  }

  interface FunctionWithError<T, R> {
    R apply(T t) throws Exception;
  }

  private <T, R> R handleError(FunctionWithError<T, R> wrapped, Function<T, R> onError, T input,
      Supplier<String> logMessage) throws Exception {
    try {
      return wrapped.apply(input);
    } catch (ServerErrorException e) {
      logException("Sam is down, returning %s".formatted(logMessage.get()), e);
      return onError.apply(input);
    }
  }
  public List<ResourceType> getResourceTypes(AuthUser authUser) throws Exception {
    return handleError(samDAO::getResourceTypes, samDefaults::getDefaultResourceTypes, authUser,
        () -> "default resource types");
  }
```

And then revert the changes to `SamDAO`. Returning the default values in `SamService` makes it easier to generically intercept each API call, since at that point the result is a java type, not a HTTP response object.

Reply:
Thank you, that makes more sense than what I thought you were suggesting - it looks promising 👍🏽